client/actions/AuthAction.js

```diff
@@ -0,0 +1,91 @@
+import axios from 'axios';
+import jwtDecode from 'jwt-decode';
+import omit from 'lodash/omit';
+
+import types from './actionTypes';
+import setAuthorizationToken from '../utils/authorization';
+
+/**
+ * isUserExists to check if passed parameter already exist in the db
+ * using the route GET /api/users/:identifier
+ *
+ * @export
+ * @param {any} identifier
+ * @returns {objeect} uuser
+ */
+export function isUserExists(identifier) {
+  return () => axios.get(`/users/verify/${identifier}`);
+}
+
+/**
+ * action dispatched on creating new user success
+ *
+ * @export
+ * @param {any} user
+ * @returns {Object} json object
+ */
+export function createUserSuccess(user) {
+  return {
+    type: types.CREATE_USER_SUCCESS,
+    user
+  };
+}
+/**
+ * action dispatched on creating new user success
+ *
+ * @export
+ * @param {any} user
+ * @returns {Object} json object
+ */
+export function setCurrentUser(user) {
+  return { type: types.SET_CURRENT_USER, user };
+}
+/**
+ * SignUp - SignUp Action
+ * @param  {object} userDetails contains the user details
+ * @return {object} return an object
+ */
+export function saveUserDetails(userDetails) {
+  return dispatch => axios.post('/users', userDetails)
+    .then((result) => {
```

Comment:
Awesome job bro, but you need to handle all your promises and request incase they fail, all `then` blocks should have a `catch` block.